scripts/bump_version.sh

```diff
@@ -0,0 +1,261 @@
+fetch_latest_version() {
+    local tag_regex='^v[0-9]\.[0-9]\.[0-9]$'
+
+    PREV_TAG=$(git describe --tags --abbrev=0)
+    PREV_TAGGED_COMMIT=$(git rev-list -n 1 "${PREV_TAG}")
+    echo "latest tag found: ${PREV_TAG} (commit ${PREV_TAGGED_COMMIT})"
+
+    if ! [[ ${PREV_TAG} =~ ${tag_regex} ]] ; then
+        echo "error: invalid tag ${PREV_TAG}" >&2
+        exit 1
+    fi
+
+    PREV_MAJOR=${PREV_TAG:1:1}
+    PREV_MINOR=${PREV_TAG:3:1}
+    PREV_PATCH=${PREV_TAG:5:1}
+
+    MAJOR=${PREV_MAJOR}
+    MINOR=${PREV_MINOR}
+    PATCH=${PREV_PATCH}
+}
+
+# Check that the working directory doesn't have un-committed changes. If it
+# does, error out.
+check_workdir_is_clean() {
+    if [ -z "$(git status --untracked-files=no --porcelain)" ]; then 
+        echo "git working directory is clean, continuing"
+    else 
+        echo "git working directory is dirty, aborting" 2>&1
+        exit 1
+    fi
+}
+
+# A helper function for interactively asking the users whether the script
+# should continue or not
+ask_yes_or_no() {
+    select yn in "Yes" "No"; do
+        case $yn in
+            Yes )
+                echo "continuing"
+                break
+                ;;
+            No )
+                echo "aborting" 2>&1
+                exit 1
+                ;;
+        esac
+    done
+}
+
+# Print a message explaining what the script does, and how to undo the changes
+# if necessary
+disclaimer() {
+    cat << EOF
+***********************************
+        IMPORTANT
+***********************************
+
+This script modifies the git commit history. If anything goes wrong, or if you
+have a doubt, you can always rollback to where this script start by running:
+
+    git reset --hard ${HEAD}
+
+This script will:
+
+1. Find the latest tag on the current branch
+2. Make sure that the CHANGELOG.md file was updated since this tag was pushed
+3. Update the version number in various files in the repository, and commit
+   these changes
+4. Create a new annotated tag
+
+EOF
+}
+
+# Print a help message
+usage() {
+    cat << EOF
+./bump_version.sh [-h|--help] [-M|--major] [-m|--minor] [-p|--patch]
+
+bump_version.sh is used for bumping the previous version number and creating a
+new tag.
+
+OPTIONS:
+
+    -h|--help:
+        print this help message
+
+    -M|--major: 
+        bump the major version number
+
+    -m|--minor: 
+        bump the minor version number
+
+    -p|--patch: 
+        bump the patch version number
+EOF
+}
+
+# Make sure the CHANGELOG was updated, and ask the user to double check the
+# changes
+check_changelog_was_updated() {
+    diff() {
+        git --no-pager diff "${PREV_TAGGED_COMMIT}" HEAD CHANGELOG.md
+    }
+
+    if [ "$(diff | wc -l)" -eq 0 ] ; then
+        echo "error: the CHANGELOG has not been updated since ${PREV_TAG}" 2>&1
+    fi
+
+    echo "The CHANGELOG has been updated since ${PREV_TAG}"
+    diff
+    echo "Do the change above look correct for v$(version)"
```

Comment:
```suggestion
    echo "Does the change above look correct for v$(version)"
```

Reply:
or maybe it's "Do the changes above ..."

Reply:
Thanks I made the change locally :)